### ISSUE TITLE
Performance loss with Julia 0.6

### ISSUE THREAD

**Issue by dpo** [JuliaSmoothOptimizers/Krylov.jl#33]
Example on matrix bcsstk18 from the UFL collection in MatrixMarket format:
```
julia> @benchmark xs, stats = cg_lanczos_shift_seq(A, b, [1,2,3])
BenchmarkTools.Trial: 
  memory estimate:  22.68 GiB
  allocs estimate:  908205
  --------------
  minimum time:     7.600 s (15.76% GC)
  median time:      7.600 s (15.76% GC)
  mean time:        7.600 s (15.76% GC)
  maximum time:     7.600 s (15.76% GC)
  --------------
  samples:          1
  evals/sample:     1

julia> VERSION
v"0.5.2"
```
but
```
julia> @benchmark xs, stats = cg_lanczos_shift_seq(A, b, [1,2,3])
BenchmarkTools.Trial: 
  memory estimate:  23.06 GiB
  allocs estimate:  2949084
  --------------
  minimum time:     10.638 s (12.18% GC)
  median time:      10.638 s (12.18% GC)
  mean time:        10.638 s (12.18% GC)
  maximum time:     10.638 s (12.18% GC)
  --------------
  samples:          1
  evals/sample:     1

julia> VERSION
v"0.6.0"
```

**Comment by dpo** [comment]
Julia 0.5.2 flame graph: https://gist.github.com/ea5614f77396287aaa790cee925585b1
Julia 0.6.0 flame graph: https://gist.github.com/dc23fa1e05c7c42b3c6fe6a046dd1929
(right-click on the image and select "open image in new tab").

**Comment by dpo** [comment]
Fixed by #32.
